src/ESP32PWM.h

```diff
@@ -24,6 +24,13 @@
 #define USABLE_ESP32_PWM (NUM_PWM-PWM_BASE_INDEX)
 #include <cstdint>
 
+#if !defined(ESP_ARDUINO_VERSION)
```

Comment:
no, please do not do this, this will cause major problems in dependant libraries that use the standard version checking to check what toolchain is compiling the code. Most of the changes you made that depend on this need to be changed back before merging.

Reply:
Sorry for being late, but GitHup missed to enable notifications for this PR for me 😞 

You are totally right! 
I moved the code to ESP32PWM.cpp.